cmd/speccheck/extended_types.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"encoding/json"
+
+	openrpc "github.com/open-rpc/meta-schema"
+)
+
+// ErrorObject represents a single error in an error group
+type ErrorObject struct {
```

Comment:
I think you might need to have an ErrorObjectOrRef as well as ErrorGroupOrRef.
for referennce for error object: 
https://github.com/open-rpc/meta-schema/blob/master/openrpc_document.go#L467



this would then account for 

x-error-groups: [ [{"$ref": "#/GasError1"}, {"$ref":"#/BlockError2"}], [{ code: 9999, message: "foobar"}], {"$ref":"#/AllUndefinedErrors} ]

where AllUndefinedErrors is of the type ErrorGroup

Reply:
Follow up comment here is that it looks like you want to throw on discovering refs, which makes sense.